pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/kit/ClinicalKitDao.java

```diff
@@ -107,4 +130,80 @@ private void setAccessionTimeForSMID(String smIdValue) {
             throw new RuntimeException("Error updating accession time for smId " + smIdValue, results.resultException);
         }
     }
+
+    public void ifTissueAccessionedTriggerDDP(String ddpParticipantId, DDPInstance ddpInstance) {
+        if (hasTissueAccessioned(ddpParticipantId, ddpInstance)) {
+            triggerParticipantEvent(ddpInstance, ddpParticipantId, DBConstants.REQUIRED_SAMPLES_RECEIVED_EVENT);
+        }
+    }
+
+    public boolean hasTissueAccessioned(String ddpParticipantId, DDPInstance ddpInstance) {
```

Comment:
I thought our general pattern was to always pass in `conn` so that callers could better manage transaction scope.  This method and `setAccessionTimeForSMID` both hide the fact that they are making single transaction updates, which means that callers may accidentally spawn multiple transactions and leave the database in an inconsistent state if their intent was to do all-or-nothing updates.

Reply:
I agree but this is how the bsp routes where historically made and I don't want for that change to refactor the whole handling of the connection now.
We should at one point revisit these things in DSM but I think that would take too much time at the moment